relays/bin-substrate/src/cli/encode_message.rs

```diff
@@ -63,15 +55,36 @@ pub(crate) fn encode_message<Source: Chain, Target: Chain>(
 ) -> anyhow::Result<RawMessage> {
 	Ok(match message {
 		Message::Raw { ref data } => data.0.clone(),
-		Message::Sized { ref size } => match *size {
-			ExplicitOrMaximal::Explicit(size) => vec![42; size as usize],
-			ExplicitOrMaximal::Maximal => {
-				let maximal_size = compute_maximal_message_size(
+		Message::Sized { ref size } => {
+			let expected_xcm_size = match *size {
+				ExplicitOrMaximal::Explicit(size) => size,
+				ExplicitOrMaximal::Maximal => compute_maximal_message_size(
 					Source::max_extrinsic_size(),
 					Target::max_extrinsic_size(),
+				),
+			};
+
+			// there's no way to craft XCM of the given size - we'll be using `ExpectPallet`
+			// instruction, which has byte vector inside
+			let mut current_vec_size = expected_xcm_size;
```

Comment:
That's the only new code in this PR - since we are now only sending messages using XCM pallet, then the message shall be valid xcm and `vec[42; 1024]` won't work here.